network/basics/uri.py

```diff
@@ -129,6 +129,13 @@
         "Content-Type" along with your request with a value of "application/json".
     required: false
     default: null
+  headers:
+    description:
+      - A dictionary of headers you wish to pass to the URI.  This can compliment or be
+        used in place of the HEADER_ method.
```

Comment:
Hmmm.  I think the HEADER_ version is a bit hacky and not very obvious.  Maybe we should have the dict version override the HEADER_ version in lieu of deprecating the HEADER_ version down the road.  I'll leave that up to the core team to decide before I make any changes.  Thanks!